packages/core/src/typeschema/types.ts

```diff
@@ -17,6 +19,7 @@ export interface InternalTypeSchema {
 }
 
 export interface ElementValidator {
+  elementDefinition: ElementDefinition;
```

Comment:
This addition is at odds with the purpose of creating `InternalTypeSchema`, which was to completely encapsulate the data from `ElementDefinition` so it didn't need to be included directly.  Which pieces of data would need to be added for parity?

Reply:
Well, to be frank, this is why I was opposed to adding a new type here in the first place.

The point is that we already have 1000's of lines of code using `ElementDefinition`, which is a perfectly standardized data type.  I can't say off the top of my head which properties we use.  We use it all over the place (`<ResourceTable>` and `<BackboneElementDisplay>` to display properties, `<ResourceForm>` and `<BackboneElementInput>` to generate inputs, `@medplum/generator` to generate type definitions and migrations).

We ***could*** rewrite everything to a new data type.  Or we ***could*** leave all of that existing code alone, which is well tested and stable in production.

If anything, I'm inclined to go the opposite direction, and revert everything to the standardized `ElementDefinition`.

A key part of the Medplum strategy is "stick to standards".  This is a mini case study in why.  If we stick to standard data types, we reduce the risk of rewriting things.

Reply:
The central issue I see is that `ElementDefinition` is not 1:1 with the totality of information for handling an element.  It works well to treat it that way in the simple limit for base resource types, but it has its warts (e.g. handling `ElementDefintion(.base).max` correctly) and in more complex scenarios with profiles the abstraction breaks down.  If we want to handle these kinds of complex cases in the React components, I think it makes sense to move more of the logic for handling the underlying FHIR schema information to one place and have as a side benefit a data structure that's simpler to work with in the ways we need to.  Rather than duplicating the data structure for `ElementDefinition`, I was hoping to explore raising the level of abstraction from `ElementDefintion` representing an individual field to accepting as a parameter only the whole type's `StructureDefinition`, which can be parsed ad hoc into the Medplum internal schema format

Reply:
I understand, and I do see merit in having a custom type.

I also want to be pragmatic.  My view is that the first order of business is to solve the double schema issue.  Updating all references to `ElementDefinition` is a non-trivial task.  I don't consider that a blocker here.